h2/src/main/org/h2/value/Value.java

```diff
@@ -481,6 +478,34 @@ public static int getHigherOrder(int t1, int t2) {
         return o1 > o2 ? t1 : t2;
     }
 
+    /**
+     * Get the higher data type of two data types. If values need to be
+     * converted to match the other operands data type, the value with the
+     * lower order is converted to the value with the higher order.
+     *
+     * @param type1 the first data type
+     * @param type2 the second data type
+     * @return the higher data type of the two
+     */
+    public static TypeInfo getHigherType(TypeInfo type1, TypeInfo type2) {
+        int t1 = type1.getValueType(), t2 = type2.getValueType();
+        if (t1 == Value.UNKNOWN || t2 == Value.UNKNOWN) {
+            if (t1 == t2) {
+                throw DbException.get(ErrorCode.UNKNOWN_DATA_TYPE_1, "?, ?");
+            } else if (t1 == Value.NULL) {
+                throw DbException.get(ErrorCode.UNKNOWN_DATA_TYPE_1, "NULL, ?");
+            } else if (t2 == Value.NULL) {
+                throw DbException.get(ErrorCode.UNKNOWN_DATA_TYPE_1, "?, NULL");
+            }
+        }
+        int dataType =  getOrder(t1) > getOrder(t2) ? t1 : t2;
```

Comment:
seems like this should be calling getHigherOrder instead of duplicating the code?

Reply:
I'll do it.

My initial implementation had some differences, so I copied the code to modify it, but these differences were removed before this PR. I didn't realize that code is exactly the same.